### ISSUE TITLE
Uncaught TypeError: Cannot read property 'offsetWidth' of undefined

### ISSUE THREAD

**Issue by vik0803** [cjmaio/activecollab-gantt#3]
There is an issue and error is been recorded on the below line

gantt.scrollTo(i,null),gantt.$scroll_hor.scrollTop=a}else{if(e.y)return!0;var n=t.wheelDelta/-40;"undefined"==typeof t.wheelDelta&&(n=t.detail);var a=gantt.$grid_data.scrollTop+30*n;gantt.scrollTo(null,a),gantt.$scroll_ver.scrollTop=a}return t.preventDefault&&t.preventDefault(),t.cancelBubble=!0,!1})},gantt._scroll_resize=function(){if(!(this._x<20||this._y<20)){var t=this._get_grid_width(),e=this._x-t,n=this._y-this.config.scale_height,i=this.config.scroll_size+1,a=this.$task_data.offsetWidth-i,s=this.config.row_height*this._order.length,r=this._get_resize_options(),o=this._scroll_hor=r.x?!1:a>e,d=this._scroll_ver=r.y?!1:s>n;

seemes that it isnt able to find the value of this.$task_data.offsetWidth

**Comment by michaeljs1990** [comment]
```javascript
gantt.scrollTo(i, null), gantt.$scroll_hor.scrollTop = a
} else {
    if (e.y) return !0;
    var n = t.wheelDelta / -40;
    "undefined" == typeof t.wheelDelta && (n = t.detail);
    var a = gantt.$grid_data.scrollTop + 30 n;
    gantt.scrollTo(null, a), gantt.$scroll_ver.scrollTop = a
}
return t.preventDefault && t.preventDefault(), t.cancelBubble = !0, !1
})
}, gantt._scroll_resize = function() {
        if (!(this._x < 20 || this._y < 20)) {
            var t = this._get_grid_width(),
                e = this._x - t,
                n = this._y - this.config.scale_height,
                i = this.config.scroll_size + 1,
                a = this.$task_data.offsetWidth - i,
                s = this.config.row_heightthis._order.length,
                r = this._get_resize_options(),
                o = this._scroll_hor = r.x ? !1 : a > e,
                d = this._scroll_ver = r.y ? !1 : s > n;
```

**Comment by vik0803** [comment]
It seems that you have the same code added. So do you think there is an error. I have the topic titled mentioned as an error received in the browser.

**Comment by cjmaio** [comment]
There may be an error, but it's in the 3rd party Gantt library. I will check for an updated version.

What were you doing when you encountered this issue?

**Comment by vik0803** [comment]
Just I went to the Gantt Tab. Selected PROJECTS and clicked on Render Gantt and I got the error.

**Comment by vik0803** [comment]
here is some more details. 
I think the 3rd party gantt is working properly. 

When I tried the url mentioned below to get_projects via API I dont get any output. Seems that the data is coming empty whereas there are 30 projects

Router::map('get_projects', 'projects/gantt/get_projects', array('controller' => 'gantt', 'action' => 'get_projects'));

**Comment by cjmaio** [comment]
Yeah... that's not with the 3rd party gantt chart then.

Are you using any filters that may be excluding your projects? If you open up the console, is activeCollab throwing any errors when it does the ajax request to get projects?

**Comment by vik0803** [comment]
Yes so I checked out with both general option thatbis by default and then selecting project option. Also i tried with all the other option but at the end it ended up with the above mentioned error that i have shared first.

Appreciate if you can share a screenshot of yours with chart and then with the direct api call so that if needed I can see into it.
